src/result_processor.c

```diff
@@ -419,7 +419,7 @@ static int rpsortNext_innerLoop(ResultProcessor *rp, SearchResult *r) {
 
     // If there is no sorting vector, load all required fields, else, load missing fields
     if (nLoadKeys == REDISEARCH_UNINITIALIZED) {
-      if (!h->rowdata.sv) {
+      if (!h->rowdata.sv && !h->rowdata.dyn) {
```

Comment:
any chance there is a scenario where both of those lists exist but there is a need to load?

Reply:
No (not right now), but even if they were both not null then we just check what field are remained to be loaded